ibm/service/kubernetes/resource_ibm_container_vpc_cluster.go

```diff
@@ -1216,6 +1222,40 @@ func waitForVpcClusterOneWorkerAvailable(d *schema.ResourceData, meta interface{
 	return createStateConf.WaitForState()
 }
 
+func waitForVpcClusterState(d *schema.ResourceData, meta interface{}, waitForState string) (interface{}, error) {
+	targetEnv, err := getVpcClusterTargetHeader(d, meta)
+	if err != nil {
+		return nil, err
+	}
+	csClient, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return nil, err
+	}
+	clusterID := d.Id()
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{deployRequested, deployInProgress},
+		Target:  []string{normal},
+		Refresh: func() (interface{}, string, error) {
+			clusterInfo, clusterInfoErr := csClient.Clusters().GetCluster(clusterID, targetEnv)
+
+			if err != nil || clusterInfoErr != nil {
+				return clusterInfo, deployInProgress, clusterInfoErr
+			}
+
+			if clusterInfo.State == strings.ToLower(waitForState) {
+				return clusterInfo, waitForState, nil
+			}
+			return clusterInfo, deployInProgress, nil
```

Comment:
this means, we cannot fail fast. if the cluster ends up in a faulty state, we would wait till timeout...